test/FrequencyTimeBalancerTest.cpp

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "FrequencyTimeBalancer.hpp"
+
+#include "FrequencyLimitDetector.hpp"
+#include "MockFrequencyLimitDetector.hpp"
+#include "geopm/Exception.hpp"
+#include "geopm/Helper.hpp"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include <memory>
+
+using geopm::FrequencyTimeBalancer;
+using ::testing::_;
+using ::testing::DoubleEq;
+using ::testing::ElementsAre;
+using ::testing::Ge;
+using ::testing::Pointwise;
+using ::testing::Return;
+
+// This CPU core filter always returns false, indicating that no cores should
```

Comment:
Are other core filters going to be added, or will this change in the future?

Reply:
In this commit, `can_ignore_control_domain_indices` tests the filter. In future commits, the agent ignores cores that are not running the application.